lib/matplotlib/colors.py

```diff
@@ -854,7 +854,7 @@ def is_gray(self):
         return (np.all(self._lut[:, 0] == self._lut[:, 1]) and
                 np.all(self._lut[:, 0] == self._lut[:, 2]))
 
-    def _resample(self, lutsize):
+    def resample(self, lutsize):
```

Comment:
I'm tempted to name this `resampled()`, which IMHO makes it more clear that it returns a new colormap and does not modify in place. This is also consistent with `Colormap.reversed()`.

Reply:
I agree on this.  Should also help make clear this is returning a new color map not mutating the one we have.